Rename API.AI References with Dialog

FabioRosado opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #293


## Status
**READY** | ~~UNDER DEVELOPMENT~~ | ~~ON HOLD~~


## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation (fix or adds documentation)


# How Has This Been Tested?

- [x] Ran all unit tests - all tests passed


# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

jacobtomlinson commented on docs/matchers/overview.md:
```diff
@@ -4,9 +4,9 @@ When writing skills for opsdroid there are multiple matchers you can use for mat
- * [API.AI](matchers/api.ai)
-    * `opsdroid.skills.match_apiai_action`
-    * `opsdroid.skills.match_apiai_intent`
+ * [Dialogflow](matchers/dialogflow)
```
Maybe this should say `Dialogflow (api.ai)`

jacobtomlinson commented on docs/matchers/dialogflow.md:
```diff
@@ -17,47 +17,47 @@ The above skill would be called on any intent which has an action of `'mydomain.
 ## Message object additional parameters
 
-### `message.apiai`
+### `message.dialogflowi`
```
You've got an `i` on the end here.

jacobtomlinson commented on opsdroid/parsers/dialogflow.py:
```diff
@@ -1,4 +1,4 @@
-"""A helper function for parsing and executing api.ai skills."""
+"""A helper function for parsing and executing Dialogflowi skills."""
```
That trailing `i` again

jacobtomlinson commented on opsdroid/parsers/dialogflow.py:
```diff
@@ -9,8 +9,8 @@
-async def call_apiai(message, config):
-    """Call the api.ai api and return the response."""
+async def call_dialogflow(message, config):
+    """Call the Dialogflowi api and return the response."""
```
That `i` again 😂

jacobtomlinson commented on docs/matchers/dialogflow.md:
```diff
@@ -1,13 +1,13 @@
 
-@match_apiai_action('mydomain.myaction')
+@match_dialogflow_action('mydomain.myaction')
 async def mySkill(opsdroid, config, message):
```
I know I wrote this example originally but it should be `my_skill` not `mySkill`, could you change it while you're at it?

jacobtomlinson commented on docs/matchers/dialogflow.md:
```diff
@@ -17,47 +17,47 @@ The above skill would be called on any intent which has an action of `'mydomain.
 
-@match_apiai_intent('myIntent')
+@match_dialogflow_intent('myIntent')
 async def mySkill(opsdroid, config, message):
```
Same as above

jacobtomlinson commented on docs/matchers/dialogflow.md:
```diff
@@ -1,13 +1,13 @@
-# api.ai Matcher
+# Dialogflow Matcher (previously api.ai)
```
According to [wikipedia](https://en.wikipedia.org/wiki/Dialogflow) the correct spelling is `Api.ai` with the capital `A` at the beginning. This could be a good opportunity to get this consistent throughout.

jacobtomlinson commented on docs/matchers/dialogflow.md:
```diff
@@ -1,13 +1,13 @@
+# Dialogflow Matcher (previously api.ai)
 
-[api.ai](https://api.ai) is an NLP API for matching strings to [intents](https://docs.api.ai/docs/concept-intents) or [actions](https://docs.api.ai/docs/concept-actions). Intents are created on the api.ai website.
+[Dialogflow](https://dialogflow.com/) is an NLP API for matching strings to [intents](https://dialogflow.com/docs/intents) or [actions](https://dialogflow.com/docs/concept-actions). Intents are created on the dialogflow website.
```
`Dialogflow` is a proper noun and should always be capitalised in documentation (but not code).

jacobtomlinson commented on docs/matchers/dialogflow.md:
```diff
@@ -17,47 +17,47 @@ The above skill would be called on any intent which has an action of `'mydomain.
 ## Configuring opsdroid
 
-In order to enable api.ai skills you must specify an `access-key` for your bot (referred to as an agent in api.ai) in the matchers section of the opsdroid configuration file. You can find this `access-key` in your agent settings. Currently you may only register one agent per opsdroid instance to avoid making multiple API calls for every message.
+In order to enable Dialogflow skills you must specify an `access-key` for your bot (referred to as an agent in dialogflow) in the matchers section of the opsdroid configuration file. You can find this `access-key` in your agent settings. Currently you may only register one agent per opsdroid instance to avoid making multiple API calls for every message.
```
Capitalize Dialogflow

jacobtomlinson commented on docs/matchers/dialogflow.md:
```diff
@@ -17,47 +17,47 @@ The above skill would be called on any intent which has an action of `'mydomain.
 
-@match_apiai_action('smalltalk.greetings')
+@match_dialogflow_action('smalltalk.greetings')
 async def dumpResponse(opsdroid, config, message):
```
Again this should be `dump_response`

jacobtomlinson commented on mkdocs.yml:
```diff
@@ -13,7 +13,7 @@ pages:
     - Overview: 'matchers/overview.md'
     - Regular Expressions: 'matchers/regex.md'
-    - API.AI: 'matchers/api.ai.md'
+    - Dialogflow (API.AI): 'matchers/dialogflow.md'
```
`Api.ai`

jacobtomlinson commented on opsdroid/matchers.py:
```diff
@@ -40,7 +43,34 @@ def match_apiai_intent(intent):
+
+
+def match_dialogflow_action(action):
+    """Return apiai action match decorator."""
```
`Api.ai`

jacobtomlinson commented on opsdroid/matchers.py:
```diff
@@ -40,7 +43,34 @@ def match_apiai_intent(intent):
+
+
+def match_dialogflow_intent(intent):
+    """Return apiai intent match decorator."""
```
`Api.ai`

jacobtomlinson commented on opsdroid/matchers.py:
```diff
@@ -40,7 +43,34 @@ def match_apiai_intent(intent):
+def match_dialogflow_intent(intent):
+    """Return apiai intent match decorator."""
+    def matcher(func):
+        """Add decorated function to skills list for apiai matching."""
```
`Api.ai`

jacobtomlinson commented on opsdroid/configuration/example_configuration.yaml:
```diff
@@ -41,7 +41,7 @@ welcome-message: true
 #     enabled: true
 #
-#   - name: apiai
+#   - name: dialogflow
```
This will need updating in the `README` too.

FabioRosado commented:
I've learned two things with this:
- I hate i's
- I shouldn't add a "READY" message to a PR when I've been awake for 16 hours haha

I think I fixed the issues that you pointed out, thanks for doing that @jacobtomlinson, I've also updated the README example(totally forgot about it before).

jacobtomlinson commented:
Many thanks!
